sql/src/main/java/org/apache/druid/sql/calcite/schema/MetadataSegmentView.java

```diff
@@ -201,32 +197,41 @@ private void poll()
       query = "/druid/coordinator/v1/metadata/segments?includeOvershadowedStatus&" + sb;
     }
     Request request;
+    InputStreamFullResponseHolder responseHolder;
     try {
       request = coordinatorClient.makeRequest(
           HttpMethod.GET,
-          StringUtils.format(query),
-          false
+          StringUtils.format(query)
       );
+
+      responseHolder = coordinatorClient.go(
+          request,
+          new InputStreamFullResponseHandler()
+      );
+      if (responseHolder.getStatus().getCode() != HttpServletResponse.SC_OK) {
+        throw new RE(
+            "Failed to talk to coordinator leader at [%s]. Error code[%d], description[%s].",
+            query,
```

Comment:
instead of `query`, it should be `request.getUrl().toString()` or something which gives info about the leader location.

Reply:
as explained above, since we change the server on-the-fly as explained above, request.getUrl() would contain  the original cached host and might print host1 which has  disappeared and response has really come from current discovered leader inside go(..) method impl, "request" object passed is not modified.
with some hackery, we could get it to print real server contacted, but I thought that would be overkill  for this logging use case and just printing that got this blah response from "coordinator leader" was good enough.